Mathlib/LinearAlgebra/DirectSum/Finsupp.lean

```diff
@@ -24,8 +24,8 @@ open DirectSum
 
 open Set LinearMap Submodule
 
-variable {R : Type u} {M : Type v} {N : Type w} [Ring R] [AddCommGroup M] [Module R M]
-  [AddCommGroup N] [Module R N]
+variable {R : Type u} {M : Type v} {N : Type w} [CommSemiring R] [AddCommMonoid M] [Module R M]
+  [AddCommMonoid N] [Module R N]
```

Comment:
This looks like it became less general in some directions? You added a commutativity requirement that wasn't here before.

Reply:
Ha! This line was totally useless here, all subsequent lemmas reintroduce assumptions… (Phew…)

Reply:
I deleted these two lines…